app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt

```diff
@@ -5,32 +5,77 @@
 package org.mozilla.focus.widget
 
 import android.content.Context
+import android.text.TextUtils
 import android.util.AttributeSet
 import android.view.LayoutInflater
+import android.view.View
 import android.widget.LinearLayout
+import kotlinx.android.synthetic.main.browser_overlay.view.*
 import org.mozilla.focus.R
+import org.mozilla.focus.autocomplete.UrlAutoCompleteFilter
+import org.mozilla.focus.utils.ViewUtils
 
 class BrowserNavigationOverlay @JvmOverloads constructor(
         context: Context, attrs: AttributeSet? = null, defStyle: Int = 0 )
-    : LinearLayout(context, attrs, defStyle) {
+    : LinearLayout(context, attrs, defStyle), View.OnClickListener {
 
     enum class NavigationEvent {
         HOME, SETTINGS, BACK, FORWARD, RELOAD, LOAD
     }
 
     interface NavigationEventHandler {
-        fun onEvent(event: NavigationEvent, value: String? = null)
+        fun onEvent(event: NavigationEvent, value: String? = null,
+                    autocompleteResult: InlineAutocompleteEditText.AutocompleteResult? = null)
     }
 
-    var eventHandler: NavigationEventHandler? = null
+    private var eventHandler: NavigationEventHandler? = null
 
     init {
         LayoutInflater.from(context)
                 .inflate(R.layout.browser_overlay, this, true)
+        listOf(navButtonBack, navButtonForward, navButtonReload, navButtonHome, navButtonSettings)
+                .forEach { setOnClickListener(this) }
+        setupUrlInput()
+    }
+
+    private fun setupUrlInput() {
+        navUrlInput.imeOptions = navUrlInput.imeOptions or ViewUtils.IME_FLAG_NO_PERSONALIZED_LEARNING
```

Comment:
I think we can allow personalized learning - I think we just don't do it for Focus